scripts/RelayHubGasEstimation.ts

```diff
@@ -505,122 +567,101 @@ const estimateGas = async () => {
+      validUntilTime: 0,
+      index: 1,
+      recoverer: constants.AddressZero,
+    },
+    relayData: {
+      callForwarder: factory.address,
+      callVerifier: verifier.address,
+      feesReceiver: relayWorker.address,
+      gasPrice: GAS_PRICE,
+    },
+  } as DeployRequest;
 
-    const { forwarderInitialBalance, relayWorkerInitialBalance } =
-      await getCurrentBalances();
+  const { signature } = await getSuffixDataAndSignature(
+    factory,
+    deployRequest,
+    owner
+  );
 
-    const transferReceiver = ethers.Wallet.createRandom();
+  const txResponse = await relayHub
+    .connect(relayWorker)
+    .deployCall(deployRequest, signature, { gasPrice: GAS_PRICE });
 
-    // necessary to execute the transfer tx without relay
-    const balanceToTransfer = 1000;
+  const { gasUsed } = await txResponse.wait();;
 
-    // forge the request
-    const completeReq: RelayRequest = await forgeRequest(
-      transferReceiver.address,
-      balanceToTransfer,
-      fees,
-      token, relayRequest,
-      forwarder
-    );
+  console.log('Total gas used on deploy: ', gasUsed.toString());
+}
 
-    const { signature } = await getSuffixDataAndSignature(
-      forwarder,
-      completeReq,
-      owner
-    );
+const estimateGas = async () => {
+  [relayWorker, relayManager, relayOwner, fundedAccount, relayHubSigner] =
+    (await ethers.getSigners()) as [
+      SignerWithAddress,
+      SignerWithAddress,
+      SignerWithAddress,
+      SignerWithAddress,
+      SignerWithAddress
+    ];
+
+  logTitle('Gas prediction - with token payment:');
+  await deployAndSetup();
+  await runGasPrediction(22, false);
 
-    const relayCallResult = await relayHub
-      .connect(relayWorker)
-      .relayCall(completeReq, signature, {
-        gasLimit,
-        gasPrice,
-      });
-    const txReceipt = await relayCallResult.wait();
-
-    await assertRelayedTransaction(
-      forwarderInitialBalance,
-      relayWorkerInitialBalance,
-      BigNumber.from(balanceToTransfer),
-      BigNumber.from(fees)
-    );
-    await printGasStatus(txReceipt);
-  }
+  logTitle('Gas prediction - without token payment:');
+  await runGasPrediction(77, true);
 
-  logTitle('Gas estimation tests for token transfer - with token payment')
-  await estimateTokenTransferGasOverhead('5000');
+  logTitle('Gas estimation tests for SmartWallet');
 
-  logTitle('Gas estimation tests for token transfer - without token payment')
-  await estimateTokenTransferGasOverhead('0');
+  ({ recipient, forwarder, owner, relayRequest, relayHub } =
+    await deployAndSetup());
 
-}
+  await runGasEstimation();
+
+  logTitle('Gas estimation scenarios');
+
+  await deployAndSetup();
+  await runGasEstimationScenarios();
+
+  logTitle('Deploy estimation without token payment');
+  await runDeployEstimation('0');
+  console.log('\nROUND 2');
+  await runDeployEstimation('0');
+
+  logTitle('Deploy estimation with token payment');
+  const TOKEN_AMOUNT = '100000000000000000000';
```

Comment:
I think we could use [the `ethers.util.parseUnit` function](https://docs.ethers.org/v5/api/utils/display-logic/#utils-parseUnits) to make this number more readable and use the ether representation. 
```suggestion
  const TOKEN_AMOUNT = ethes.utils.parseUnit('100');
```

Reply:
Done.